daemon/src/projection.rs

```diff
@@ -341,21 +339,25 @@ impl Cfd {
 
                 self.state = CfdState::PendingClose;
             }
-            CollaborativeSettlementRejected { commit_tx } => {
-                self.aggregated.commit_tx = Some(commit_tx);
+            CollaborativeSettlementRejected { .. } => {
+                self.aggregated.commit_published = true;
                 self.pending_settlement_proposal_price = None;
```

Comment:
not from this PR, but it caught my attention: why do we assume that commit got published after the collab settlement is rejected? AFAICT it's a higher-level business decision that the taker does, the projection should not be aware of it?

Reply:
We started doing this when we merged event sourcing. 
The idea is: If the taker wants to close the taker wants to close. If the maker rejects we force close. 
We can question this again, but I think it is fair to say that once the taker hits close the taker wants the position to close.

Reply:
I don't mind living with that business decision, but it seems odd and fragile to encode it in this place? I would assume that we'd rather emit the force close event in the cfd actor after a failure/rejection? (or maybe first retry after failure?)

Reply:
It is a fair question. The way I see it is that the modelling of the event encodes this business decision. The fact that it includes the commit transaction means we want it to get published. So it is not so much the projection doing this reasoning, it is the projection adhering to the contract that the event provides.

Reply:
It is part of the event logic to do that atm.
I created a discussion for it, because I'm actually not sure about it: https://github.com/itchysats/itchysats/discussions/1132

Reply:
> It is a fair question. The way I see it is that the modelling of the event encodes this business decision. The fact that it includes the commit transaction means we want it to get published. So it is not so much the projection doing this reasoning, it is the projection adhering to the contract that the event provides.

This is an interesting point, but doesn't it also mean that our business decisions become *immutable* as a result? thanks @da-kami for creating a discussion, when funding rate is in, I aim to catch up on all the discussions ;D

Reply:
@thomaseizinger please feel free to push this, these remarks are as such unrelated to the PR, and they might as well go in 👍